### PR TITLE
add: xhr-mock

### PR DESCRIPTION
**Pull request by joscha**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/xhr-mock/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        post(url: string, fn: mock.MockFunction): this;
+        put(url: string, fn: mock.MockFunction): this;
+        patch(url: string, fn: mock.MockFunction): this;
+        delete(url: string, fn: mock.MockFunction): this;
+    }
+}
+
+declare var mock: mock.XhrMock;
+export = mock;
+export as namespace mock;
```
Comment (minestarks) [inline types/xhr-mock/index.d.ts]: This is to enable usage as a UMD module (i.e. a global). This is also what your tests use. But is this actually allowed by the library? The documentation only shows `require()` usage.

**Comment by minestarks** [comment]
Thanks for the PR. Would like to get confirmation about that `export as namespace` question before I merge.

File: types/xhr-mock/index.d.ts (hunk 1 above)
Comment (joscha) [reply types/xhr-mock/index.d.ts]: @minestarks - it was the only way I could make the synthetic interfaces (request, response) available in the target code. I wasn't aware this also meant exposure as a global, so I think you are right and it is incorrect. Do you have a suggestion about how to make the interfaces available instead?

Comment (minestarks) [reply types/xhr-mock/index.d.ts]: Once you've imported the module via `import mock = require('xhr-mock');`, you should be able to reference your interfaces from user code like `mock.MockResponse` and `mock.MockResponse`.

Comment (joscha) [reply types/xhr-mock/index.d.ts]: fixed in c88276a0d

**Comment by joscha** [comment]
changes in @minestarks

**Comment by joscha** [comment]
Ah @typescript-bot, you are a funny one, aren't you, will have a look in a sec ;)

**Review by minestarks — APPROVED** [review]

**Pull request merged**